### PR TITLE
Add the possibility to change mail list view into full width

### LINKED ISSUE #8811: Full-width mail list
## Feature Request

As a Nextcloud Mail user I may prefer to see as much information about a messages/thread as possible before I open it. The current view, for example, shows only a small part of the subject and preview text:

Screenshot 2023-08-25 at 14 37 17

## Describe the solution you'd like

Allow me to switch to a full-width message/thread list. In other words, the three column design (nav + list + content) becomes two column: nav + list or nav + content. The user will see the list. When they navigate to a message/thread, the content shows.

To switch between the views we need a three way button or actions menu. This button is places next to the button for advanced search.

## References

This is the trend in some of the major email apps as UX people realize that the idea of multitasking was a bit overrated.

| Product | Screenshot |
|--------|--------|
| Hey  | Screenshot 2023-08-25 at 15 29 18 | 
| Gmail | Screenshot 2023-08-25 at 15 22 46 |
| Spark Mail | Screenshot 2023-08-25 at 15 25 36 | 

## Implementation

Right now Mail uses the following structure:

```html
<NcContent>
  <Navigation />
  <AppContent>
    <div slot="list">
      <SearchMessages />
      <AppContentList>
        <Mailbox />
      </AppContentList>
    </div>
    <AppContentDetails />
  </AppContent>
</NcContent>
```

*Ideally*, we can keep the structure as-is, but enhance AppContent and ListItem to work with one or two columns:

<table>
<tr>
<th>List + content</th>
<th>List only</th>
<th>Content only</th>
</tr>
<tr>
<td>

```html
<NcContent>
  <Navigation />
  <AppContent layout="vertical-split">
    <div slot="list">
      <SearchMessages />
      <AppContentList>
        <ListItem layout="compact" />
      </AppContentList>
    </div>
    <AppContentDetails />
  </AppContent>
</NcContent>
```

</td>
<td>

```html
<NcContent>
  <Navigation />
  <AppContent layout="list">
    <div slot="list">
      <SearchMessages />
      <AppContentList>
        <ListItem layout="wide" />
      </AppContentList>
    </div>
    <AppContentDetails /><!-- not rendered -->
  </AppContent>
</NcContent>
```

</td>
<td>

```html
<NcContent>
  <Navigation />
  <AppContent layout="content">
    <div slot="list"><!-- not rendered -->
      <SearchMessages />
      <AppContentList>
        <ListItem layout="wide" />
      </AppContentList>
    </div>
    <AppContentDetails />
  </AppContent>
</NcContent>
```

</td>
</tr>
</table>

Long code snipped short, we don't control the rendering, we just tell  AppContent and ListItem the layout to render.

This gives us the following advantages
1) We do not have to rip apart the ListItem with deep selectors or write a completely new component.
2) The new layouts streamlines mobile view with desktop view.
3) Other apps can use this too, e.g. to offer a better mobile view or a layout switch, if desired.
4) It will be easy to implement https://github.com/nextcloud/mail/issues/8309

Challenges
* The table(-like) structure of AppContentList and ListItem for the list-only mode and potential wrapping components between (currently Mailbox, EnvelopeList and transition)

### PR DESCRIPTION
**Pull request by GretaD**
Fix #8811 

- [x] Requires https://github.com/nextcloud-libraries/nextcloud-vue/pull/5209
- [x] Requires https://github.com/nextcloud-libraries/nextcloud-vue/pull/5391

Screenshot from 2024-03-11 11-07-38

### CODE REVIEW

File: src/components/AppSettingsMenu.vue [hunk 1]
```diff
@@ -4,6 +4,26 @@
...
+					@change="toggleFullWidthLayout(true) ">
+					<template #icon>
+						<CompactMode :size="20" />
+					</template>
+					{{ t('mail', 'Show full mode') }}
+				</NcActionRadio>
+				<NcActionRadio :value="'default-view'"
+					:name="t('mail', 'Default view')"
+					:checked="isDefaultView"
+					@change="toggleFullWidthLayout('default-view') ">
```
Comment (st3iny) [inline src/components/AppSettingsMenu.vue:20]:
```suggestion
					:checked="!isFullWidthLayout"
					@update:checked="setFullWidthLayout(false)">
```

**Comment by GretaD** [comment]
oneline layout (yes, i know the important icon is not palced correctly)
Screenshot from 2024-02-28 16-14-12
vertical split layout
Screenshot from 2024-02-28 16-12-58

**Comment by ChristophWurst** [comment]
Noticed during testing
- [ ] Back button always causes the envelope list to reload. This seems unnecessary and makes interactions slow.
- [ ] Sidebar toggle and back button have inconsistent color (black vs gray) and do not align horizontally.
- [ ] Important icon is broken in list mode
- [ ] The clickable area and hover area of envelopes is small in list mode. Increase padding and reduce margin at the same rate to make the elements bigger without taking more space.

Other
- Subject/preview is not styled as columns. The longer the subject, the less I see of the preview. This is by design spec but it looks strange.

**Review by st3iny — COMMENTED** [review]
- [x] 1. There are two scroll bars when using list layout.
- [ ] 2. Scrolling to load more doesn't work on portrait screens
    - This is a know and wont-fix bug of the old layout as well. We just hide it by loading enough messages so that the list is always scrollable. Now, the list fits more items so we should increase the amount of initial envelopes a bit in list mode.

### About 1.)
grafik

File: src/components/Envelope.vue [hunk 2]
```diff
@@ -1047,10 +1047,6 @@ export default {
 	display: flex;
 	width: 200px;
 }
-:deep(.list-item--one-line) {
-	padding: 10px;
-	margin: 2px;
-}
```
Comment (st3iny) [inline src/components/Envelope.vue]: Oops, that was fix for the clickable area. I'll push it again.

File: src/components/Envelope.vue [hunk 3]
```diff
@@ -903,6 +915,9 @@ export default {
 	}
 	}
 }
+.important-one-line.app-content-list-item-star:deep() {
+	top: 1px;
```
[the remark this answers is not in the document]
Comment (st3iny) [reply src/components/Envelope.vue]: Done

File: src/components/Envelope.vue (hunk 2 above)
Comment (st3iny) [reply src/components/Envelope.vue]: Hmm, I should learn to read :see_no_evil: The changes are already upstreamed. I'll revert again.

**Comment by hamza221** [comment]
Current state not indicate, Maybe it should be like this?
| B | A |
|--------|--------|
| image | image |

File: src/components/AppSettingsMenu.vue [hunk 4]
```diff
@@ -4,6 +4,28 @@
 			:name="t('mail', 'Mail settings')"
 			:show-navigation="true"
 			:open.sync="showSettings">
+			<NcAppSettingsSection id="mail-list-view" :name="t('mail', 'Layout')">
+				<NcCheckboxRadioSwitch value="no-split"
+					name="mail-layout"
+					type="radio"
+					:checked="layoutMode"
```
Comment (ChristophWurst) [inline src/components/AppSettingsMenu.vue:11]:
```suggestion
					:checked="layoutMode === 'no-split"
```

to revive the checked state?

**Review by ChristophWurst — APPROVED** [review]
No more double scrolling bar :+1: 

image

whitespace on top  of thread header can be improved

**Comment by ChristophWurst** [comment]
:shipit:

**Comment by GretaD** [comment]
> whitespace on top of thread header can be improved

i will do this in a follow up pr, after the horizontal layout is in as well, because for sure it will need some changes there too

**Review by hamza221 — APPROVED** [review]

**Pull request merged**